### DISCUSSION TITLE
0 instances predicted (no instances found) after labeling

### DISCUSSION THREAD

**Discussion by hawkinso** [talmolab/sleap#1595] [Help!] [answered]
I am trying to train a dataset that has 9 points placed along a flapping foil. I have labeled 42 random frames and I am trying to run a training test that only iterates through 2 epochs. Each time I run it, **I get the following error:** 

![error](https://github.com/talmolab/sleap/assets/91697828/c3ee6035-2901-4712-a25e-f8343cc77a3d)

**This is the output from command terminal:** 
![terminalerror](https://github.com/talmolab/sleap/assets/91697828/c0671cd0-cb25-4e83-83c7-5013527cb8f0)

**Here is what I specify in the pipeline:** 

![TrainingPipeline](https://github.com/talmolab/sleap/assets/91697828/3cca2cd9-b575-4fb1-9f4f-d9a9e3376535)

**Here is what I specify for the Single Instance Model Configuration:** 
![SingleInstanceModelConfig](https://github.com/talmolab/sleap/assets/91697828/60410466-e2d5-4fc5-8293-1f9990d6e723)

I am fairly new to working with SLEAP so I am not sure if I am missing certain parameters or if this is a labeling issue. Even when I labeled 230 frames over six videos I still received the same error. To label for this particular training set I used the random sample method in which I labeled 21 random frames from two test videos. I am currently using SLEAP version 1.3.3. 

Any help would be very appreciated!

**Comment by talmo** [comment, marked as answer]
Hi @hawkinso,

Thanks for reaching out!

If the model fails to optimize for long enough, it'll be unable to generate good predictions, so it won't find any instances in your frames during inference.

Definitely the most obvious culprit is the number of epochs -- I'd recommend setting this to the default 200. SLEAP will stop earlier once it detects that the model training converged.

Other than that, I see that the images appear to be quite large! I'd recommend a few tweaks to the configuration you screenshotted above:

- Input scaling: 0.5
- Epochs: 200
- Max stride: 32 (or maybe 64)
- Filters: 24
- Filters rate: 1.5
- Output stride: 4 (if 2 doesn't work or you get memory issues)

Give those a go and let us know if you run into any issues!

Cheers,

Talmo

**Reply by hawkinso** [reply]
This worked perfectly! Thank you so much for your help!
